pkg/controllers/monitoring-stack/controller.go

```diff
@@ -130,20 +131,74 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	availableCondition := metav1.Condition{
+		Type:               "Available",
+		ObservedGeneration: ms.ObjectMeta.Generation,
+		LastTransitionTime: metav1.Time{Time: time.Now().UTC()},
+	}
+
 	reconcilers := stackComponentReconcilers(ms, r.instanceSelectorKey, r.instanceSelectorValue)
 	for _, reconciler := range reconcilers {
 		err := reconciler(ctx, r.k8sClient, r.scheme)
 		// handle create / update errors that can happen due to a stale cache by
 		// retrying after some time.
 		if errors.IsAlreadyExists(err) || errors.IsConflict(err) {
 			logger.V(3).Info("skipping reconcile error", "err", err)
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
```

Comment:
We shouldn't set "Failure" here since like the comment says, we are ignoring `AlreadyExists` and `Conflict` errors.

Reply:
OK

Reply:
Hum so what should we return in this case please? Requeue as well with `ctrl.Result{RequeueAfter: 2 * time.Second}, nil` or do not update the status at all?

Reply:
We should preserve the old behaviour. we shouldn't set any status here not return error because we are just ignoring an error due to stale cache.